### PR TITLE
LOGIST-659 Extract client

### PR DESCRIPTION
**Pull request by nenadstojanovikj**
This PR extracts the communication with Google Cloud Service into separate Client.

### CODE REVIEW

File: src/GoogleCloudTracer/Client/ClientInterface.php [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+{
+    /**
+     * @link https://cloud.google.com/trace/docs/reference/v1/rest/v1/projects/patchTraces
+     *
+     * @param string $traceJson JSON payload which is going to be sent to Google Cloud Traces
+     *
+     * @return bool
+     */
+    public function patchTraces(string $traceJson) : bool;
+}
```
Comment (boekkooi-fresh) [inline src/GoogleCloudTracer/Client/ClientInterface.php]: should we make the auth request part of the client?

Comment (nenadstojanovikj) [inline src/GoogleCloudTracer/Client/ClientInterface.php]: I think the separation is fine this way.

Comment (boekkooi-fresh) [inline src/GoogleCloudTracer/Client/ClientInterface.php]: Then please rename this from `ClientInterface` into something else. Maybe `RecorderClientInterface`?

**Comment by boekkooi-fresh** [comment]
:+1:

**Comment by emilyclaudet** [comment]
👍

**Comment by Shru-10** [comment]
+1

**Pull request merged**